Adding generic setup configuration to the README.md (#716)
jenkinsci/kubernetes-plugin: README.md

see the [Docker image source code](https://github.com/jenkinsci/docker-jnlp-slave).

It is not required to run the Jenkins master inside Kubernetes. 
# Generic Setup
### Prerequisites
* A running Kubernetes cluster
* An instances of Jenkins installed
* The Jenkins Kubernetes plugin installed

Fill in the Kubernetes plugin configuration. In order to do that, you will open the Jenkins UI and navigate to 

> “Manage Jenkins -> Configure System -> Cloud -> Kubernetes” and enter in the ‘Kubernetes URL’ and ‘Jenkins URL’ 

Reviewer: Unless Jenkins is running in Kubernetes in which case the defaults work.

Author: I will add that point

> appropriately. To test this connection is successful you can use the `Test Connection` button to ensure there is 

Reviewer: ```suggestion
appropriately. To test this connection is successful you can use the **Test Connection** button to ensure there is 
```

adequate communication from Jenkins to the Kubernetes cluster, as seen below

![image](Screen%20Shot%202020-02-20%20at%202.00.28%20PM.png)

> In addition to that, in the ‘Kubernetes Pod Template’ section, we need to configure the image that will be used to 

Reviewer: ```suggestion
In addition to that, in the **Kubernetes Pod Template** section, we need to configure the image that will be used to 
```

spin up the agent pod. If you have some custom requirements for your agent, you can build one more dockerfile with the 
appropriate changes the same way we did for the Jenkins control plane. On the other hand, if you don’t have unique 
requirements 

> for your agent, you can use the default Jenkins agent image available on the official Docker hub repository. In the 

Reviewer: This is all pretty off base I think.

First of all, we do not recommend overriding the `jnlp` container except under unusual circumstances. You pretty much always just use the default one. For actual build steps, you use separate containers, which can be anything, and the `container` step.

Second, it is generally discouraged to even use the global pod template configuration. Better to use the `podTemplate` step and define your requirements directly inside the job—or via some library etc.

The existing samples, for example in #707, show these things in action. No global configuration or custom images are required.

Author: Okay, I will remove that in favor of the base `jnlp` container.
As for the `podTemplate`, would it make sense to give the user (in terms of steps) both options?

Reviewer: I guess. The main reason to use the global pod template definition is to migrate a huge corpus of existing projects (incl. freestyle) to run on Kubernetes without changing job definitions. New users setting up new Kubernetes builds should use the `podTemplate` step as shown in the example snippets.

Author: okay, I will add that line of wording.
Thank you for helping

see the [Docker image source code](https://github.com/jenkinsci/docker-jnlp-slave).

It is not required to run the Jenkins master inside Kubernetes. 
# Generic Setup
### Prerequisites
* A running Kubernetes cluster

> * An instances of Jenkins installed

Reviewer 2: ```suggestion
* A Jenkins instance installed
```

* The Jenkins Kubernetes plugin installed

> It first should be noted that the main reason to use the global pod template definition is to migrate a huge corpus of 

Reviewer 2: ```suggestion
It should be noted that the main reason to use the global pod template definition is to migrate a huge corpus of 
```

existing projects (incl. freestyle) to run on Kubernetes without changing job definitions. New users setting up new 
Kubernetes builds should use the podTemplate step as shown in the example snippets [here](https://github.com/jenkinsci/kubernetes-plugin/pull/707)

Fill in the Kubernetes plugin configuration. In order to do that, you will open the Jenkins UI and navigate to 
“Manage Jenkins -> Configure System -> Cloud -> Kubernetes” and enter in the ‘Kubernetes URL’ and ‘Jenkins URL’ 
appropriately, this is unless Jenkins is running in Kubernetes in which case the defaults work. To test this connection is successful you can use the **Test Connection** button to ensure there is 
adequate communication from Jenkins to the Kubernetes cluster, as seen below

![image](Screen%20Shot%202020-02-20%20at%202.00.28%20PM.png)

In addition to that, in the **Kubernetes Pod Template** section, we need to configure the image that will be used to 
spin up the agent pod. We do not recommend overriding the `jnlp` container except under unusual circumstances. 

> for your agent, you can use the default Jenkins agent image available on the official Docker hub repository. In the 

Reviewer 2: ```suggestion
for your agent, you can use the default Jenkins agent image available in [Docker Hub](https://hub.docker.com). In the 
```

‘Kubernetes Pod Template’ section you need to specify the following (the rest of the configuration is up to you):
Kubernetes Pod Template Name - can be any and will be shown as a prefix for unique generated agent’ names, which will 
be run automatically during builds
Docker image - the docker image name that will be used as a reference to spin up a new Jenkins agent, as seen below

> ![image](Screen%20Shot%202020-02-20%20at%202.01.48%20PM.png)

Reviewer: Can we rename these images please, so that the filenames are descriptive and do not use dangerous characters?

Reviewer 2: Already done.
